### PR TITLE
Added notebook demonstrating distributed hyperparameter tuning

### PR DESCRIPTION
**Pull request by nikitamaia**
Notebook demonstrating distributed hyperparameter tuning on Vertex Training.

### CODE REVIEW

File: ai-platform-unified/notebooks/unofficial/hyperparameter_tuning/distributed-hyperparameter-tuning.ipynb [hunk 1]
```diff
@@ -0,0 +1,879 @@
+{
```
Comment (andrewferlitsch) [inline ai-platform-unified/notebooks/unofficial/hyperparameter_tuning/distributed-hyperparameter-tuning.ipynb:1]: <p>In Overview, refer to this as the "mirrored strategy" for distributed training.</p><p> </p><p>There are two Vertex hyperparameter tuning services. Be specific which one is used here.</p><p> </p><p>Inconsistent capitalization:</p><p>Vertex AI Hyperparameter</p><p>Vertex AI hyperparameter</p><p> </p><p>Except when referring to Vertex AI as a platform, the AI part is dropped.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/ai-platform-samples/pull/391/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (andrewferlitsch) [inline ai-platform-unified/notebooks/unofficial/hyperparameter_tuning/distributed-hyperparameter-tuning.ipynb:1]: <p>Keras -&gt; TF.Keras</p><p> </p><p>this library -&gt; the hypertune library</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/ai-platform-samples/pull/391/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='34'/>

Comment (andrewferlitsch) [inline ai-platform-unified/notebooks/unofficial/hyperparameter_tuning/distributed-hyperparameter-tuning.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #19.      parser.add_argument(
</pre><p>Convention is "units" not "num_neurons"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/ai-platform-samples/pull/391/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='35'/>

Comment (andrewferlitsch) [inline ai-platform-unified/notebooks/unofficial/hyperparameter_tuning/distributed-hyperparameter-tuning.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #85.      history = model.fit(
</pre><p>number of epochs should be argument: --epochs</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/ai-platform-samples/pull/391/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='35'/>

Comment (andrewferlitsch) [inline ai-platform-unified/notebooks/unofficial/hyperparameter_tuning/distributed-hyperparameter-tuning.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #5.    ! docker build ./ -t $IMAGE_URI
</pre><p>./ seems confusing. I would be the container code one subfolder down. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/ai-platform-samples/pull/391/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='38'/>

Comment (andrewferlitsch) [inline ai-platform-unified/notebooks/unofficial/hyperparameter_tuning/distributed-hyperparameter-tuning.ipynb:1]: <p>Change to present tense.</p><p>After your container has been pushed to Google Container Registry -&gt;</p><p>Once your container is pushed to Google Container Registry</p><p> </p><p>You define the following specs -&gt; You define the following specifications</p><p> </p><p>that has two&nbsp;<code>NVIDIA_TESLA_T4</code>&nbsp;GPUs. -&gt; with two&nbsp;<code>NVIDIA_TESLA_T4</code>&nbsp;GPUs.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/ai-platform-samples/pull/391/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='41'/>

Comment (andrewferlitsch) [inline ai-platform-unified/notebooks/unofficial/hyperparameter_tuning/distributed-hyperparameter-tuning.ipynb:1]: <p>Puts -&gt; Sets</p><p> </p><p>It is a best practice to start with -&gt; The recommended practice is to start with</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/ai-platform-samples/pull/391/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='45'/>

Comment (andrewferlitsch) [inline ai-platform-unified/notebooks/unofficial/hyperparameter_tuning/distributed-hyperparameter-tuning.ipynb:1]: <p>missing the argument for search algorithm (set it to the default None)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/ai-platform-samples/pull/391/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='46'/>

Comment (andrewferlitsch) [inline ai-platform-unified/notebooks/unofficial/hyperparameter_tuning/distributed-hyperparameter-tuning.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #11.    hp_job.run()
</pre><p>What's the results. How do you interpret them.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/ai-platform-samples/pull/391/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='46'/>

Comment (andrewferlitsch) [inline ai-platform-unified/notebooks/unofficial/hyperparameter_tuning/distributed-hyperparameter-tuning.ipynb:1]: <p>you only cleaned up the bucket, not the hyperparam tuning job</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/ai-platform-samples/pull/391/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='48'/>

**Comment by andrewferlitsch** [comment]
I've reviewed the notebook, see ReviewNB for comments

File: ai-platform-unified/notebooks/unofficial/hyperparameter_tuning/distributed-hyperparameter-tuning.ipynb (hunk 1 above)
Comment (nikitamaia) [reply ai-platform-unified/notebooks/unofficial/hyperparameter_tuning/distributed-hyperparameter-tuning.ipynb:1]: Done

Comment (nikitamaia) [reply ai-platform-unified/notebooks/unofficial/hyperparameter_tuning/distributed-hyperparameter-tuning.ipynb:1]: Done

Comment (nikitamaia) [reply ai-platform-unified/notebooks/unofficial/hyperparameter_tuning/distributed-hyperparameter-tuning.ipynb:1]: Done

Comment (nikitamaia) [reply ai-platform-unified/notebooks/unofficial/hyperparameter_tuning/distributed-hyperparameter-tuning.ipynb:1]: Done

Comment (nikitamaia) [reply ai-platform-unified/notebooks/unofficial/hyperparameter_tuning/distributed-hyperparameter-tuning.ipynb:1]: Done

Comment (nikitamaia) [reply ai-platform-unified/notebooks/unofficial/hyperparameter_tuning/distributed-hyperparameter-tuning.ipynb:1]: Done

Comment (nikitamaia) [reply ai-platform-unified/notebooks/unofficial/hyperparameter_tuning/distributed-hyperparameter-tuning.ipynb:1]: Done

Comment (nikitamaia) [reply ai-platform-unified/notebooks/unofficial/hyperparameter_tuning/distributed-hyperparameter-tuning.ipynb:1]: No model is created, just a training pipeline. Does that need to be deleted as well? I don't think training pipelines result in extra costs if they are not deleted. If it does need to be deleted, do you know how to do that with gsutil?

Comment (nikitamaia) [reply ai-platform-unified/notebooks/unofficial/hyperparameter_tuning/distributed-hyperparameter-tuning.ipynb:1]: Done.

**Comment by ivanmkc** [comment]
This notebook is great to have. Thanks for contributing.

**Review by ivanmkc — APPROVED** [review]
LGTM

**Pull request merged**